packages/devextreme/testing/tests/DevExpress.ui.widgets.scheduler/common.events.tests.js

```diff
@@ -718,4 +718,49 @@ QUnit.module('Events', {
 
         assert.equal(contentReadyFiresCount, 3, 'contentReadyFiresCount === 3');
     });
+
+    QUnit.test('onAppointmentContextMenu should be triggered when items in the appointment tooltip and appointment collector tooltip is right-clicked (T1181442)', function(assert) {
+        let $eventAppointmentElement = null;
+        const onAppointmentContextMenu = sinon.spy(({ appointmentElement }) => {
+            $eventAppointmentElement = $(appointmentElement);
+        });
+
+        createWrapper({
+            height: 600,
+            dataSource: [{
+                text: 'appointment 1',
+                startDate: new Date(2023, 7, 18),
+                endDate: new Date(2023, 7, 18, 4),
+            }, {
+                text: 'appointment 2',
+                startDate: new Date(2023, 7, 18),
+                endDate: new Date(2023, 7, 18, 4),
+            }],
+            maxAppointmentsPerCell: 1,
+            currentDate: new Date(2023, 7, 18),
+            onAppointmentContextMenu,
+        });
+
+        const $appointment = $('.dx-scheduler-appointment');
+        $appointment.trigger('dxclick');
+
+        this.clock.tick(300);
+
+        let $appointmentItem = $('.dx-list-item');
```

Comment:
All this code is hidden in the createWrapper object. See examples in source code and declaration this object

Reply:
refactored